db/lib/types.js

```diff
@@ -10,6 +10,7 @@ const PayloadRetrievabilityState = {
 const PayloadRetrievabilityStateType = Type.Enum(PayloadRetrievabilityState)
 
 const ActiveDealDbEntry = Type.Object({
+  id: Type.Optional(Type.Number()),
```

Comment:
What would be missing to make this type required (which should be safer)?

Reply:
Am I right in my understanding that we need this to be optional so that when we create a new entry from the block event (in backend/lib/utils.js) we can keep it empty?

Reply:
If we want to be strict about types, then there should be two types:
1. "NewActiveDealDbEntry" that does not have any ID value - we want the database to generate the value for us
2. "ActiveDealDbEntry" where the ID value is required (always present)

Reply:
In TypeScript, it's possible to create a new type from another type by omitting certain properties - see https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys.

```ts
type NewActiveDealDbEntry = Omit<ActiveDealDbEntry, "id">
```

That obviously does not provide runtime validation, so if we want to go down the route of two types, then we need to find a solution based on Typebox features.

Reply:
@juliangruber you are correct, if we want to have the type clarity before inserting deals into the database (like we are doing in `backend/lib/utils.js`), we need to keep the id field optional. We do not know the id of a deal before inserting it. The id also does not have any meaning outside of the context of the database. 
The only scenario I see where this could become an issue is if we try to look for a deal using the id and the typing from `Typebox` without knowing the id. 
I do not think it makes much of a difference whether we use typebox during runtime to assert the existence of the id or whether we simply check it using an if statement wherever necessary.

Reply:
> if we want to have the type clarity before inserting deals into the database (like we are doing in backend/lib/utils.js), we need to keep the id field optional. 

On the contrary, the optional id field does not give type clarity. If we wanted to have type clarify, we would need separate types, one for with ID and one for without, so that at any time you know with certainty whether an object has the ID or not.

I'm fine not adding a second type, if we find a better name for the type. The type is currently called `ActiveDealDbEntry`. Such a type must have an `id`, as we can't have a db entry without an id (due to the schema). Therefore, this type gives a false sense of security.

Reply:
I see your point. Let me split it up then into a deal type before inserting into the database and a deal type that corresponds to the data being read from the database.

Reply:
I see you added `ActiveDeal` type for data without the auto-generated `id` property 👍🏻 

I would expect the `id` property of `ActiveDealDbEntry` to be required now. Do you agree?

```suggestion
  id: Type.Number(),
```